File: guest/c/aslr_probe.c
// aslr_probe.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int stack_var = 42;
    void *heap_ptr = malloc(16);

    printf("PROBE_OK\n");
    printf("MAIN=%p\n", (void *)&main);
    printf("STACK=%p\n", (void *)&stack_var);
    printf("HEAP=%p\n", heap_ptr);

    free(heap_ptr);
    return 0;
}

File: guest/rust/aslr_probe.rs
fn main() {
    let stack_var: i32 = 42;
    let heap_ptr = Box::new(1234);

    println!("PROBE_OK");
    println!("MAIN={:p}", main as *const ());
    println!("STACK={:p}", &stack_var as *const i32);
    println!("HEAP={:p}", &*heap_ptr as *const i32);
}
